client/src/components/common/SlidesLayout.tsx

```diff
@@ -0,0 +1,68 @@
+import React, {useEffect, useState} from 'react';
+import Pagination from "./Pagination";
+import Cards from "./Cards";
+import slideApi, {SlideResponse} from "../../api/slide";
+import {googleAnalyticsPageView} from "../../utils/googleAnalytics";
+import styled from "@emotion/styled";
+
+const SlidesBlock = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  height: 100%;
+`;
+
+interface IProps {
+  slidesCnt: number
+  title: string
+}
+
+const SlidesLayout: React.FC<IProps> = ({slidesCnt, title}) => {
+  const [slides, setSlides] = useState<Array<SlideResponse>>([]);
+  const [page, setPage] = useState<number>(0);
+  const [totalPage, setTotalPage] = useState<number>(1);
+
+  useEffect(() => {
+    slideApi.getAll({page, size: slidesCnt})
+      .then(res => {
+        setSlides(res.data.slides);
+        setTotalPage(res.data.totalPage);
+      });
+  }, [page, totalPage]);
+
+  const onClickPage = (e: React.MouseEvent<HTMLDivElement>) => {
+    const page = parseInt(e.currentTarget.getAttribute("data-page")!);
+    setPage(page);
+    slideApi.getAll({page, size: slidesCnt})
+      .then(res => setSlides(res.data.slides));
+  };
+
+  const onClickPrev = (prev: number) => {
+    setPage(prev);
+    slideApi.getAll({page, size: slidesCnt})
+      .then(res => setSlides(res.data.slides));
+  };
+
+  const onClickNext = (next: number) => {
+    setPage(next);
+    slideApi.getAll({page, size: slidesCnt})
+      .then(res => setSlides(res.data.slides));
+  };
```

Comment:
```suggestion
  const onClick = (pageNumber: number) => {
    setPage(pageNumber);
    slideApi.getAll({page, size: slidesCnt})
      .then(res => setSlides(res.data.slides));
  };
```
여기를 합치면 될 것 같은데,,???